### PR TITLE
[v-hacd] Add new port

### PR DESCRIPTION
**Pull request by dhkatz**
**Describe the pull request**

- What does your PR fix? Adds https://github.com/kmammou/v-hacd

- Which triplets are supported/not supported? Have you updated the CI baseline? Supports all triplets but ARM.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

### CODE REVIEW

**Comment by dhkatz** [comment]
I'm not really sure why the build fails on arm64 or uwp. I'd like to get it working if possible through build options. If not then add them as unsupported. I have no way of testing those directly myself.

**Comment by NancyLi1013** [comment]
@dhkatz 
This port currently should not support `arm`.
The errors like this:
```
4>C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.25.28610\include\emmintrin.h(24,1): fatal error C1189: #error:  This header is specific to X86 and X64 targets (compiling source file C:\agent\_work\1\s\buildtrees\v-hacd\src\e17ceb9aef-3a5e115dd0\src\VHACD_Lib\src\btAlignedAllocator.cpp) [C:\agent\_work\1\s\buildtrees\v-hacd\arm-uwp-dbg\src\VHACD_Lib\vhacd.vcxproj]
     4>C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.25.28610\include\emmintrin.h(24,1): fatal error C1189: #error:  This header is specific to X86 and X64 targets (compiling source file C:\agent\_work\1\s\buildtrees\v-hacd\src\e17ceb9aef-3a5e115dd0\src\VHACD_Lib\src\btConvexHullComputer.cpp) [C:\agent\_work\1\s\buildtrees\v-hacd\arm-uwp-dbg\src\VHACD_Lib\vhacd.vcxproj]
     4>C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.25.28610\include\emmintrin.h(24,1): fatal error C1189: #error:  This header is specific to X86 and X64 targets (compiling source file C:\agent\_work\1\s\buildtrees\v-hacd\src\e17ceb9aef-3a5e115dd0\src\VHACD_Lib\src\vhacdMesh.cpp) [C:\agent\_work\1\s\buildtrees\v-hacd\arm-uwp-dbg\src\VHACD_Lib\vhacd.vcxproj]
     4>C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.25.28610\include\emmintrin.h(24,1): fatal error C1189: #error:  This header is specific to X86 and X64 targets (compiling source file C:\agent\_work\1\s\buildtrees\v-hacd\src\e17ceb9aef-3a5e115dd0\src\VHACD_Lib\src\vhacdVolume.cpp) [C:\agent\_work\1\s\buildtrees\v-hacd\arm-uwp-dbg\src\VHACD_Lib\vhacd.vcxproj]
```
So you can add the unsupported info.

**Comment by dhkatz** [comment]
@NancyLi1013 Everything seems good now

File: ports/v-hacd/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,44 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```
Comment (NancyLi1013) [inline ports/v-hacd/portfile.cmake:1]:
```suggestion
vcpkg_fail_port_install(ON_ARCH "arm")

vcpkg_check_linkage(ONLY_STATIC_LIBRARY)

```
We still need to add this here.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Need to test features.

**Comment by NancyLi1013** [comment]
All features have passed with the following triplets:

- x86-windows
- x64-windows
- x64-windows-static

File: ports/v-hacd/fix-cmake.patch [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+ )
+ 
+-export(EXPORT vhacd-targets NAMESPACE ::
+-  FILE "${CMAKE_CURRENT_BINARY_DIR}/vhacd/vhacd-targets.cmake"
++export(EXPORT vhacd-targets NAMESPACE vhacd::
++  FILE "${CMAKE_CURRENT_BINARY_DIR}/vhacd/v-hacd-targets.cmake"
+ )
+ 
+ configure_file(cmake/vhacd-config.cmake
+-  "${CMAKE_CURRENT_BINARY_DIR}/vhacd/vhacd-config.cmake"
```
Comment (dan-shaw) [inline ports/v-hacd/fix-cmake.patch:91]: Not sure why we are renaming config targets

**Comment by dan-shaw** [comment]
Thanks for the PR!

**Pull request merged**